modules/statistics-provider-influx/src/main/java/org/opencastproject/statistics/provider/influx/provider/InfluxTimeSeriesStatisticsProvider.java

```diff
@@ -0,0 +1,112 @@
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.statistics.provider.influx.provider;
+
+import org.opencastproject.statistics.api.DataResolution;
+import org.opencastproject.statistics.api.ResourceType;
+import org.opencastproject.statistics.api.TimeSeries;
+import org.opencastproject.statistics.api.TimeSeriesProvider;
+import org.opencastproject.statistics.provider.influx.StatisticsProviderInfluxService;
+import org.opencastproject.util.data.Tuple;
+
+import org.influxdb.dto.BoundParameterQuery;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class InfluxTimeSeriesStatisticsProvider extends InfluxStatisticsProvider implements TimeSeriesProvider {
+
+  private String aggregation;
+  private String aggregationVariable;
+  private String measurement;
+  private String resourceIdName;
+
+
+  public InfluxTimeSeriesStatisticsProvider(
+      StatisticsProviderInfluxService service,
+      String id,
+      ResourceType resourceType,
+      Set<DataResolution> dataResolutions,
+      String title,
+      String description,
+      String aggregation,
+      String aggregationVariable,
+      String measurement,
+      String resourceIdName
+  ) {
+    super(service, id, resourceType, dataResolutions, title, description);
+    this.aggregation = aggregation;
+    this.aggregationVariable = aggregationVariable;
+    this.measurement = measurement;
+    this.resourceIdName = resourceIdName;
+  }
+
+  @Override
+  public TimeSeries getValues(String resourceId, Instant from, Instant to, DataResolution resolution, ZoneId zoneId) {
+    final String influxGrouping = dataResolutionToInfluxGrouping(resolution);
+    final List<Tuple<Instant, Instant>> periods = getPeriods(from, to, resolution, zoneId);
+    final List<String> labels = new ArrayList<>();
+    final List<Double> values = new ArrayList<>();
+    for (final Tuple<Instant, Instant> period : periods) {
+      final Query query = BoundParameterQuery.QueryBuilder
+              .newQuery("SELECT " + aggregation + "(" + aggregationVariable + ") FROM " + measurement + " WHERE "
+                                + resourceIdName + "=$resourceId AND time>=$from AND time<=$to" + influxGrouping)
+              .bind("resourceId", resourceId)
+              .bind("from", period.getA())
+              .bind("to", period.getB())
+              .create();
+      final QueryResult results = service.getInfluxDB().query(query);
```

Comment:
`influxDB.query(…)` may fail on some conditions. Especially if the Influx DB is not running or not reachable. It would make sense to throw an exception in that case and mark the method as "throws exceptions" in the API.

Reply:
Actually, `queryResultToTimeSeries()` checks for errors and throws an exception if any. I would like to leave that a `RuntimeException` since checked exceptions don't go well with java8's functional features (which are used in `StatisticsEndpoint`).

Reply:
Unfortunately true. I will loose my `chuck(e)` war in opencast only because exceptions are not intended in functional programming.
And no, `queryResultToTimeSeries()` does not filter the case, the connection to influx db is blocked by the firewall. You will see a really ugly and long stacktrace with a simple reason:

```
Caused by: org.influxdb.InfluxDBIOException: java.net.ConnectException: Failed to connect to /127.0.0.1:8086
```

Is there a way to handle this case properly? You can easy reproduce it by stopping influx db and call the /statistics/timeseries endpoint.

Reply:
I have added special handling for connect exceptions. In this case, now only the error message is logged.